libraries/tools/kotlin-gradle-plugin-integration-tests/src/test/kotlin/org/jetbrains/kotlin/gradle/KotlinGradlePluginIT.kt

```diff
@@ -1053,4 +1053,39 @@ class KotlinGradleIT : BaseGradleIT() {
             assertTasksExecuted(":lib1:compileDebugKotlin")
         }
     }
+
+    /** Regression test for KT-38692. */
+    @Test
+    fun testIncrementalWhenNoKotlinSources() = with(
+        Project("kotlinProject")
+    ) {
+        setupWorkingDir()
+        assertTrue(this.allKotlinFiles.toList().isNotEmpty())
+        build(":compileKotlin") {
+            assertSuccessful()
+            assertTasksExecuted(":compileKotlin")
+        }
+
+        // Remove all Kotlin sources and also modify buildscript classpath
+        allKotlinFiles.forEach { assertTrue(it.delete()) }
+        projectDir.resolve("src/main/java/Sample.java").also {
+            it.parentFile.mkdirs()
+            it.writeText("public class Sample {}")
+        }
+        gradleBuildScript().modify {
```

Comment:
nit: `--rerun-tasks` may be simpler? (to avoid adding `org.testng:testng:6.8` which makes it slightly difficult to upgrade that library)

Reply:
Nice, I forgot about `--rerun-tasks` :)